trinity/extensibility/plugin.py

```diff
@@ -6,17 +6,43 @@
     ArgumentParser,
     _SubParsersAction,
 )
+from enum import (
+    auto,
+    Enum,
+)
 import logging
 from typing import (
-    Any
+    Any,
+    Dict,
+)
+
+from lahja import (
+    Endpoint
 )
 
 from trinity.extensibility.events import (
     BaseEvent
 )
+from trinity.utils.mp import (
+    ctx,
+)
+from trinity.utils.logging import (
+    setup_queue_logging,
+)
+
+
+class PluginProcessScope(Enum):
 
-# TODO: we spec this out later
-PluginContext = Any
+    SHARED = auto()
+    MAIN = auto()
+    OWN = auto()
+
+
+class PluginContext:
```

Comment:
I vaguely remember at least `PluginContext(object)` being recommended somewhere...

Reply:
I believe that since python3 the two are equivalent.